smartsim/_core/mli/infrastructure/environmentloader.py

```diff
@@ -42,20 +46,38 @@ class EnvironmentConfigLoader:
     """
 
     def __init__(self) -> None:
-        self._feature_store_descriptor: t.Optional[str] = os.getenv(
-            "SSFeatureStore", None
-        )
         self._queue_descriptor: t.Optional[str] = os.getenv("SSQueue", None)
-        self.feature_store: t.Optional[FeatureStore] = None
+        self.feature_stores: t.Optional[t.Dict[str, FeatureStore]] = None
         self.queue: t.Optional[DragonFLIChannel] = None
+        self._prefix = "SSFeatureStore"
+
+    def _load_feature_store(self, env_var: str) -> FeatureStore:
+        """Load a feature store from a descriptor
+        :param descriptor: The descriptor of the feature store
+        :returns: The hydrated feature store"""
+        logger.debug(f"Loading feature store from env: {env_var}")
 
-    def get_feature_store(self) -> t.Optional[FeatureStore]:
-        """Loads the Feature Store previously set in SSFeatureStore"""
-        if self._feature_store_descriptor is not None:
-            self.feature_store = pickle.loads(
-                base64.b64decode(self._feature_store_descriptor)
+        value = os.getenv(env_var)
+        if not value:
+            raise SmartSimError(
+                f"Empty feature store descriptor in environment: {env_var}"
             )
-        return self.feature_store
+
+        try:
+            return t.cast(FeatureStore, pickle.loads(base64.b64decode(value)))
+        except:
+            raise SmartSimError(
```

Comment:
Just to double check, this is a spot that we are okay raising an error in. We don't want to try to continue on if one `FeatureStore` fails to rehydrate?

Reply:
This particular error means it had a value but it was malformed (e.g. truncated, not b64 encoded, bad padding). I thought it was worth a true failure to avoid users assuming a feature store was available when it really wasn't

Reply:
removed the exceptions but now considering that return None is abused per @mellis13